scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -211,6 +219,46 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
         i
       }
       val next = if (nextPos != -1) scannerTokens(nextPos) else null
+
+      val currentIndent = {
+        if (curr.is[Whitespace]) {
+          -1
+        } else {
+          var i = currPos - 1
+          var ind = 0
+          var commentLF = false
+          def isWhitespace(token: Token): Boolean =
+            token.is[Space] || token.is[Tab] || token.is[Comment]
+          while (i >= 0 && isWhitespace(scannerTokens(i))) {
+            scannerTokens(i) match {
+              case c: Comment =>
+                val lfpos = c.value.reverse.indexOf("\n")
+                if (lfpos != -1) commentLF = true
+                ind += (if (lfpos != -1) lfpos else c.value.size)
```

Comment:
so if there are multiple comments following each other, we should add their last-line lengths? why add? and why last-line length:

```
   /* foo
    * bar */ /*
    * baz
    * qux */
```

should we just keep the 4 spaces before `* qux */`? or add `4 + "* qux */".length + 4 + "* bar */".length`?

Reply:
gonna ask dotty guys about it, I added minimal support to be able to parse dotty codebase and did not consider all scenarios

Reply:
yeah okay so comments in relation to indentation is a grey area apparently, I am going to just ignore it for now and add it to scala3 todo issue.